components/new_post/index.tsx

```diff
@@ -0,0 +1,228 @@
+
+import {
+    get,
+    getBool,
+    isCollapsedThreadsEnabled,
+} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentTeam, getCurrentTeamId, getTeam, getTeamMemberships} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentUserId, getUser} from 'mattermost-redux/selectors/entities/users';
+
+import {Emoji} from '@mattermost/types/emojis';
+import {Post} from '@mattermost/types/posts';
+import {closeRightHandSide, selectPost, selectPostCardFromRightHandSideSearch, setRhsExpanded} from 'actions/views/rhs';
+
+import {markPostAsUnread, emitShortcutReactToLastPostFrom} from 'actions/post_actions';
+
+import {getShortcutReactToLastPostEmittedFrom, getOneClickReactionEmojis} from 'selectors/emojis';
+import {getIsPostBeingEditedInRHS, isEmbedVisible} from 'selectors/posts';
+import {getHighlightedPostId, getRhsState} from 'selectors/rhs';
+import {getIsMobileView} from 'selectors/views/browser';
+
+import {GlobalState} from 'types/store';
+
+import {isArchivedChannel} from 'utils/channel_utils';
+import {areConsecutivePostsBySameUser, shouldShowActionsMenu} from 'utils/post_utils';
+import {Preferences, RHSStates} from 'utils/constants';
+
+import {ExtendedPost, removePost} from 'mattermost-redux/actions/posts';
+import {DispatchFunc, GetStateFunc} from 'mattermost-redux/types/actions';
+import {isThreadOpen} from 'selectors/views/threads';
+
+import {General} from 'mattermost-redux/constants';
+
+import {getDisplayNameByUser} from 'utils/utils';
+import {getDirectTeammate} from 'mattermost-redux/selectors/entities/channels';
+
+import PostComponent from './post_component';
+
+interface OwnProps {
+    post?: Post;
+    previousPostId?: string;
+    postId?: string;
+    teamId?: string;
+    shouldHighlight?: boolean;
+}
+
+function isFirstReply(post: Post, previousPost?: Post | null): boolean {
+    if (post.root_id) {
+        if (previousPost) {
+            // Returns true as long as the previous post is part of a different thread
+            return post.root_id !== previousPost.id && post.root_id !== previousPost.root_id;
+        }
+
+        // The previous post is not a real post
+        return true;
+    }
+
+    // This post is not a reply
+    return false;
+}
+
+function isConsecutivePost(state: GlobalState, ownProps: OwnProps) {
+    let post;
+    if (ownProps.postId) {
+        post = getPost(state, ownProps.postId);
+    } else if (ownProps.post) {
+        post = ownProps.post;
+    }
+    const previousPost = ownProps.previousPostId && getPost(state, ownProps.previousPostId);
+
+    let consecutivePost = false;
+
+    if (previousPost && post) {
+        consecutivePost = areConsecutivePostsBySameUser(post, previousPost);
+    }
+    return consecutivePost;
+}
+
+function removePostAndCloseRHS(post: ExtendedPost) {
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState() as GlobalState;
```

Comment:
why did we had to implicitly type it to GlobalState?

Reply:
this was a copy paste from another index file so I didn't test it, checking now, we have two types of global state, one from mm-redux and another specified in types/store. I guess we need a separate ticket to unify them

Reply:
No actually we dont have to unify them since one of them is in regards to webapp specifically eg has plugin etc.